ESP_ATMod/command.cpp

```diff
@@ -2366,22 +2485,32 @@ commands_t findCommand(uint8_t *input, uint16_t inpLen)
 	if (inpLen == 4)
 		return CMD_AT;
 
+	char *inputCmd = (char *)calloc(strlen((char *)input + 2) + 1, sizeof(char));
+	strcpy(inputCmd, (char *)input + 2);
+
 	// Check the command list
 	for (unsigned int i = 0; i < sizeof(commandList) / sizeof(commandDef_t); ++i)
 	{
 		const char *cmd = commandList[i].text;
+		String compareCmd = commandList[i].text;

```

Comment:
I think we should use another way to find a command. This may work but seems too complicated to me.
I'd prefer finding a token in an input string: `if (!memcmp(cmd, input + 2, strlen(cmd)))`
Then
a) MODE_EXACT_MATCH - input must continue with \r\n (i.e. check the length)
b) MODE_QUERY_SET - input must continue with either '?' or '='
c) MODE_NO_CHECKING - input must not continue with [A-Z_]

Isn't this better? BTW: you are not deleting the inputCmd buffer (memory leaks).
I can make the changes after merging the PR myself.

Reply:
Need to take a deeper look at this. The reason why I changed it was because it sometimes found the wrong command if that command was the shorter version of the command it was actually looking for.

Reply:
Yes, I see. The commands with the same prefix were in particular order, from the longest to the shortest.

Reply:
That is ofcourse a way of fixing it, but personally I see this as a really dirty fix. If someone tries to adjust or implement something and they make the tiny mistake to just put the command somewhere they like (like I did), it can compelety mess up everything. That's why I introduced the order and a way to check commands order independent.

But I'll take a look at this and see if there is another way to do this better, but I'm quite sure this is the best I could do by checking the command on the command itself and still keeping your modechecking in place. :)

Reply:
It was quite dirty :) but it seemed to me functional enough and (with some comment) understandable.
I think we can define the command as one of types:
- AT+COMMAND
- AT+COMMAND?
- AT+COMMAND=xxxxx

Reply:
Oh, my bad, I forgot to check this change.
There is still the problem with memory freeing when the command is not found.
I refactor this as soon as possible, I have another idea of checking for the command.